### PR TITLE
Spec presence and whoami

### PR DESCRIPTION
**Pull request by bassory99**

### CODE REVIEW

File: aiobaro/core.py [hunk 1]
```diff
@@ -995,16 +1026,38 @@ async def set_presence(
...
+        Rate-limited:	Yes.
+        Requires auth:	Yes.
         """
-        return MatrixResponse(httpx.Response(status_code=404, json={}))
+
+        return await self.auth_client(
+            "PUT",
+            f"presence/{user_id}/status",
+            json=dict(
+                filter(
```
Comment (moas) [inline aiobaro/core.py:1040]: bad pattern which avoid debug practice. it's more evidence and clean if you create a variable in this case

File: tests/test_aiobaro.py [hunk 2]
```diff
@@ -274,52 +274,89 @@ async def test_thumbnail(matrix_client):
...
-    result = await matrix_client.profile_get(*args, **kwargs)
+@pytest.mark.asyncio
+async def test_profile_set_displayname(matrix_client):
+    await test_register(matrix_client)
+    await test_login(matrix_client)
+
+    user_id = "@test_user:baro"
+    display_name = "Test USER"
+    result = await matrix_client.profile_set_displayname(user_id, display_name)
     assert result.ok
```
Comment (moas) [inline tests/test_aiobaro.py:285]: you must to assert the content response! Now don't known if response contains another user display name or not

File: tests/test_aiobaro.py [hunk 3]
```diff
@@ -274,52 +274,89 @@ async def test_thumbnail(matrix_client):
...
+@pytest.mark.asyncio
 async def test_profile_get_displayname(matrix_client):
-    args, kwargs = [], {}
-    result = await matrix_client.profile_get_displayname(*args, **kwargs)
+    await test_profile_set_displayname(matrix_client)
+
+    user_id = "@test_user:baro"
+    result = await matrix_client.profile_get_displayname(user_id)
     assert result.ok
+    assert result.json().get("displayname")
```
Comment (moas) [inline tests/test_aiobaro.py:295]: assert response contains the key but you must check the value

File: tests/test_aiobaro.py [hunk 4]
```diff
@@ -274,52 +274,89 @@ async def test_thumbnail(matrix_client):
...
-    args, kwargs = [], {}
-    result = await matrix_client.set_presence(*args, **kwargs)
+@pytest.mark.asyncio
+async def test_get_presence(matrix_client):
+    await test_set_presence(matrix_client)
+
+    user_id = "@test_user:baro"
+    result = await matrix_client.get_presence(user_id)
     assert result.ok
+    assert result.json().get("presence")
```
Comment (moas) [inline tests/test_aiobaro.py:349]: what's presence state expected ?

File: aiobaro/core.py (hunk 1 above)
Comment (bassory99) [reply aiobaro/core.py:1051]: Thanks for the tip ! We will have to fix this in more places

File: tests/test_aiobaro.py (hunk 2 above)
Comment (bassory99) [reply tests/test_aiobaro.py:285]: According to the Matrix specification, the response for this action is only "{}". We then only asserted result.ok

Comment (moas) [reply tests/test_aiobaro.py:285]: ok! all right! I mistaked in confusing set/get

File: tests/test_aiobaro.py (hunk 3 above)
Comment (bassory99) [reply tests/test_aiobaro.py:295]: Thanks, will be done!

File: tests/test_aiobaro.py (hunk 4 above)
Comment (bassory99) [reply tests/test_aiobaro.py:349]: Thanks, will be done!

**Review by oukone — APPROVED** [review]
LGTM

**Pull request merged**